src/oc_erchef/apps/chef_index/c_src/fast_xs.c

```diff
@@ -0,0 +1,210 @@
+#include <assert.h>
```

Comment:
The fast_xs code is nearly all unchanged (one change for dialzyer in the erlang side fast_xs module). As for the rest here is a summary:

- chef_index_http: unchaged
- chef_index_expand: The core expansion code was not touched, but the gen_server was removed and a small public facing API was exposed.  The gen_server in chef_index_expand was focused on batches of a known size, which we generally won't have in the case of indexing inflight requests, thus the chef_index_batch gen_server.
-